src/SFML/Graphics/Font.cpp

```diff
@@ -637,24 +631,21 @@ Glyph Font::loadGlyph(std::uint32_t codePoint, unsigned int characterSize, bool
         else
         {
             // Pixels are 8 bits gray levels
-            for (unsigned int y = padding; y < height - padding; ++y)
+            for (unsigned int y = padding; y < size.y - padding; ++y)
             {
-                for (unsigned int x = padding; x < width - padding; ++x)
+                for (unsigned int x = padding; x < size.x - padding; ++x)
                 {
                     // The color channels remain white, just fill the alpha channel
-                    const std::size_t index      = x + y * width;
+                    const std::size_t index      = x + y * size.x;
                     m_pixelBuffer[index * 4 + 3] = pixels[x - padding];
                 }
                 pixels += bitmap.pitch;
             }
         }
 
         // Write the pixels to the texture
-        const unsigned int x = static_cast<unsigned int>(glyph.textureRect.left) - padding;
-        const unsigned int y = static_cast<unsigned int>(glyph.textureRect.top) - padding;
-        const unsigned int w = static_cast<unsigned int>(glyph.textureRect.width) + 2 * padding;
-        const unsigned int h = static_cast<unsigned int>(glyph.textureRect.height) + 2 * padding;
```

Comment:
By reading the code above.
In the original code:
- `glyph.textureRect` is first set to `findGlyphRect(page, {width, height})`, which returns a rectangle of the given size (_when there is no error_). At this moment `{width, height}` is `{bitmap.width + 2 * padding, bitmap.rows + 2 * padding}`.
- Then the padding is removed. So `glyph.textureRect` is of size `{bitmap.width, bitmap.rows}`
- So when `w` and `h` are computed with padding, they are set to `{bitmap.width + 2 * padding, bitmap.rows + 2 * padding}`
- Conclusion: `page.texture.update` is called with size `{bitmap.width + 2 * padding, bitmap.rows + 2 * padding}`.

In my suggested changes:
- `size` is initialized to `{bitmap.width, bitmap.rows}`
- Then padding is added, so `size` becomes `{bitmap.width + 2 * padding, bitmap.rows + 2 * padding}`
- Conclusion: `page.texture.update` is also called with size `{bitmap.width + 2 * padding, bitmap.rows + 2 * padding}`.

I decided to pass `size` to the `update` method because `size` is used to define the size of `m_pixelBuffer`, which is what we are supposed to pass to `update`.

BUT, _when there is an error_ in `findGlyphRect` (no space left on the page texture), the previous code would copy a few transparent pixels of `m_pixelBuffer` into `page.texture` whereas my suggested changes would copy the glyph on the top left corner of the page texture.
So I need to fix my code, thank you for bringing this to my attention.

Reply:
Instead of reverting my changes to the previous behavior, I think I improved it.

To test it, I added underline style to the tennis example text (to check interaction with the white pixels in the top left corner of the page texture used for underline),
```patch
diff --git a/examples/tennis/Tennis.cpp b/examples/tennis/Tennis.cpp
index 41b05b8e..48a0c96a 100644
--- a/examples/tennis/Tennis.cpp
+++ b/examples/tennis/Tennis.cpp
@@ -95,6 +95,7 @@ int main()
     pauseMessage.setCharacterSize(40);
     pauseMessage.setPosition({170.f, 200.f});
     pauseMessage.setFillColor(sf::Color::White);
+    pauseMessage.setStyle(sf::Text::Style::Underlined);
 
 #ifdef SFML_SYSTEM_IOS
     pauseMessage.setString("Welcome to SFML Tennis!\nTouch the screen to start the game.");
```

and I forced `findGlyphRect` to fail with the following patches.

On master branch: Underline is not showing and characters that could not be copied on the page texture are invisible.
```patch
diff --git a/src/SFML/Graphics/Font.cpp b/src/SFML/Graphics/Font.cpp
index b009d65e..e69a9b49 100644
--- a/src/SFML/Graphics/Font.cpp
+++ b/src/SFML/Graphics/Font.cpp
@@ -668,6 +668,10 @@ Glyph Font::loadGlyph(std::uint32_t codePoint, unsigned int characterSize, bool
 ////////////////////////////////////////////////////////////
 IntRect Font::findGlyphRect(Page& page, const Vector2u& size) const
 {
+    static int x = 0;
+    if (x++ % 2 == 1)
+        return {{0, 0}, {2, 2}};
+
     // Find the line that fits well the glyph
     Row*  row       = nullptr;
     float bestRatio = 0;
```
![main](https://github.com/SFML/SFML/assets/1485306/58442f8c-e32b-437e-b407-1cd6bbb43c60)

On this branch: Underline is visible and characters that could not be copied on the page texture are shown in white.
```patch
diff --git a/src/SFML/Graphics/Font.cpp b/src/SFML/Graphics/Font.cpp
index d6408dba..006e2ed5 100644
--- a/src/SFML/Graphics/Font.cpp
+++ b/src/SFML/Graphics/Font.cpp
@@ -663,6 +663,10 @@ Glyph Font::loadGlyph(std::uint32_t codePoint, unsigned int characterSize, bool
 ////////////////////////////////////////////////////////////
 IntRect Font::findGlyphRect(Page& page, const Vector2u& size) const
 {
+    static int x = 0;
+    if (x++ % 2 == 1)
+        return {{0, 0}, {0, 0}};
+
     // Find the line that fits well the glyph
     Row*  row       = nullptr;
     float bestRatio = 0;
```
![branch](https://github.com/SFML/SFML/assets/1485306/177d5364-aad3-4ef6-a3f8-c6abf19851ae)

Does that look good to you?

Reply:
I am saving this for follow-up bugfix PR to keep this one focused.
https://github.com/kimci86/SFML/compare/feature/rect_as_two_vectors...kimci86:SFML:bugfix/handle_findGlyphRect_error

Reply:
Now that you're saving the proper bug fix for a later PR does that mean you need to revert your usage of `size` in the `page.texture.update` call?